Prediction from clip-ViT-B-32-multilingual-v1 incorrectly returns 768 elements rather than 512

## Description
Prediction from clip-ViT-B-32-multilingual-v1 incorrectly returns 768 elements.

### Expected Behavior

Returns 512 element vectors, as shown by this Python code:

```python
from sentence_transformers import SentenceTransformer, util
import numpy as np

model = SentenceTransformer('sentence-transformers/clip-ViT-B-32-multilingual-v1')
emb = model.encode('fast blue car')
np_array = np.array(emb)

print("Length: ")
print(len(np_array))

print("Result: ")
print(' '.join(map(str, np_array)))
```
This returns:

```
Length: 
512
Result: 
0.12500623 -0.40152556 -0.044941768 ...
```

## How to Reproduce?
Here is an associated unit test:

```java
    @Test
    public void testIssue() throws Exception
    {
        Criteria<String, float[]> criteria =
            Criteria.builder()
                    .setTypes(String.class, float[].class)
                    .optModelUrls("djl://ai.djl.huggingface.pytorch/sentence-transformers/clip-ViT-B-32-multilingual-v1")
                    .optTranslatorFactory(new TextEmbeddingTranslatorFactory())
                    .optEngine("PyTorch")
                    .build();
        try (ZooModel<String, float[]> model = criteria.loadModel();
             Predictor<String, float[]> predictor = model.newPredictor())
        {
            float[] embeddings = predictor.predict("fast blue car");
            assertThat(embeddings.length, is(512));
            assertThat((double) embeddings[0], is(closeTo(0.12500623, 0.0001)));
        }
    }
```

### Steps to reproduce
Run the above unit test.

### Thoughts

https://huggingface.co/sentence-transformers/clip-ViT-B-32-multilingual-v1/discussions/4 reports a similar issue when trying to use Optimum with this model, where apparently it was ignoring the pooling and dense layer inclusion (which reduces the dimensions to 512).  Could a similar issue be happening here?

@frankfliu - many thanks for the sample code.  Once I downloaded the `model.safetensors` file so it is accessible, I ran your code, but it only returned 1, rather than 512.  To be precise, it returned a float array with this value: `[0.0038979053]`.  Any ideas what is wrong here?

I also tried your first suggestion of tracing the model, using this code:

```python
from opensearch_py_ml.ml_models import SentenceTransformerModel

model_id = "sentence-transformers/clip-ViT-B-32-multilingual-v1"
folder_path = "/data/output/djl/clip-ViT-B-32-multilingual-v1"
pre_trained_model = SentenceTransformerModel(model_id=model_id, folder_path=folder_path, overwrite=True)
model_path = pre_trained_model.save_as_pt(model_id=model_id, sentences=["fast blue car"])
```
However, when I try an use this model from within djl-serving, I seem to get this error:

```
TRACE Predictor Predictor input data: [ND: (5) cpu() int64
[  101, 15040, 23254, 13000,   102]
, ND: (5) cpu() int64
[ 1,  1,  1,  1,  1]
]
DEBUG JniUtils Expected at most 2 argument(s) for operator 'forward', but received 3 argument(s). Declaration: forward(__torch__.sentence_transformers.SentenceTransformer.SentenceTransformer self, Dict(str, Tensor) input) -> Dict(str, Tensor)
Exception raised from checkAndNormalizeInputs at ../aten/src/ATen/core/function_schema_inl.h:392 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x6b (0x77360a6b405b in /data/nuix/djl-serving/cache/pytorch/2.0.1-cpu-linux-x86_64/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0xbf (0x77360a6aef6f in /data/nuix/djl-serving/cache/pytorch/2.0.1-cpu-linux-x86_64/libc10.so)
frame #2: void c10::FunctionSchema::checkAndNormalizeInputs<c10::Type>(std::vector<c10::IValue, std::allocator<c10::IValue> >&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, c10::IValue, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, c10::IValue> > > const&) const + 0x6ae (0x7734c1dbd03e in /data/nuix/djl-serving/cache/pytorch/2.0.1-cpu-linux-x86_64/libtorch_cpu.so)
frame #3: torch::jit::Method::operator()(std::vector<c10::IValue, std::allocator<c10::IValue> >, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, c10::IValue, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, c10::IValue> > > const&) const + 0x173 (0x7734c4dc0b53 in /data/nuix/djl-serving/cache/pytorch/2.0.1-cpu-linux-x86_64/libtorch_cpu.so)
frame #4: <unknown function> + 0x12c950 (0x77353172c950 in /data/nuix/djl-serving/cache/pytorch/2.0.1-cpu-linux-x86_64/0.26.0-libdjl_torch.so)
frame #5: Java_ai_djl_pytorch_jni_PyTorchLibrary_moduleRunMethod + 0x183 (0x77353172cbbc in /data/nuix/djl-serving/cache/pytorch/2.0.1-cpu-linux-x86_64/0.26.0-libdjl_torch.so)
frame #6: [0x7735ec80fa70]

WARN  WorkerThread Failed to predict
ai.djl.translate.TranslateException: ai.djl.engine.EngineException: Expected at most 2 argument(s) for operator 'forward', but received 3 argument(s). Declaration: forward(__torch__.sentence_transformers.SentenceTransformer.SentenceTransformer self, Dict(str, Tensor) input) -> Dict(str, Tensor)
	at ai.djl.inference.Predictor.batchPredict(Predictor.java:192) ~[api-0.26.0.jar:?]
	at ai.djl.serving.wlm.ModelInfo$ModelThread.lambda$run$1(ModelInfo.java:1088) ~[wlm-0.26.0.jar:?]
	at ai.djl.serving.wlm.Job.runAll(Job.java:68) ~[wlm-0.26.0.jar:?]
	at ai.djl.serving.wlm.ModelInfo$ModelThread.run(ModelInfo.java:1088) ~[wlm-0.26.0.jar:?]
	at ai.djl.serving.wlm.WorkerThread.runJobs(WorkerThread.java:137) ~[wlm-0.26.0.jar:?]
	at ai.djl.serving.wlm.WorkerThread.run(WorkerThread.java:87) [wlm-0.26.0.jar:?]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.base/java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: ai.djl.engine.EngineException: Expected at most 2 argument(s) for operator 'forward', but received 3 argument(s). Declaration: forward(__torch__.sentence_transformers.SentenceTransformer.SentenceTransformer self, Dict(str, Tensor) input) -> Dict(str, Tensor)
	at ai.djl.pytorch.jni.PyTorchLibrary.moduleRunMethod(Native Method) ~[pytorch-engine-0.26.0.jar:?]
	at ai.djl.pytorch.jni.IValueUtils.forward(IValueUtils.java:57) ~[pytorch-engine-0.26.0.jar:?]
	at ai.djl.pytorch.engine.PtSymbolBlock.forwardInternal(PtSymbolBlock.java:146) ~[pytorch-engine-0.26.0.jar:?]
	at ai.djl.nn.AbstractBaseBlock.forward(AbstractBaseBlock.java:79) ~[api-0.26.0.jar:?]
	at ai.djl.nn.Block.forward(Block.java:127) ~[api-0.26.0.jar:?]
	at ai.djl.inference.Predictor.predictInternal(Predictor.java:143TRACE Predictor Predictor input data: [ND: (5) cpu() int64
[  101, 15040, 23254, 13000,   102]
, ND: (5) cpu() int64
[ 1,  1,  1,  1,  1]
]
DEBUG JniUtils Expected at most 2 argument(s) for operator 'forward', but received 3 argument(s). Declaration: forward(__torch__.sentence_transformers.SentenceTransformer.SentenceTransformer self, Dict(str, Tensor) input) -> Dict(str, Tensor)
Exception raised from checkAndNormalizeInputs at ../aten/src/ATen/core/function_schema_inl.h:392 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x6b (0x77360a6b405b in /data/nuix/djl-serving/cache/pytorch/2.0.1-cpu-linux-x86_64/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0xbf (0x77360a6aef6f in /data/nuix/djl-serving/cache/pytorch/2.0.1-cpu-linux-x86_64/libc10.so)
frame #2: void c10::FunctionSchema::checkAndNormalizeInputs<c10::Type>(std::vector<c10::IValue, std::allocator<c10::IValue> >&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, c10::IValue, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, c10::IValue> > > const&) const + 0x6ae (0x7734c1dbd03e in /data/nuix/djl-serving/cache/pytorch/2.0.1-cpu-linux-x86_64/libtorch_cpu.so)
frame #3: torch::jit::Method::operator()(std::vector<c10::IValue, std::allocator<c10::IValue> >, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, c10::IValue, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, c10::IValue> > > const&) const + 0x173 (0x7734c4dc0b53 in /data/nuix/djl-serving/cache/pytorch/2.0.1-cpu-linux-x86_64/libtorch_cpu.so)
frame #4: <unknown function> + 0x12c950 (0x77353172c950 in /data/nuix/djl-serving/cache/pytorch/2.0.1-cpu-linux-x86_64/0.26.0-libdjl_torch.so)
frame #5: Java_ai_djl_pytorch_jni_PyTorchLibrary_moduleRunMethod + 0x183 (0x77353172cbbc in /data/nuix/djl-serving/cache/pytorch/2.0.1-cpu-linux-x86_64/0.26.0-libdjl_torch.so)
frame #6: [0x7735ec80fa70]

WARN  WorkerThread Failed to predict
ai.djl.translate.TranslateException: ai.djl.engine.EngineException: Expected at most 2 argument(s) for operator 'forward', but received 3 argument(s). Declaration: forward(__torch__.sentence_transformers.SentenceTransformer.SentenceTransformer self, Dict(str, Tensor) input) -> Dict(str, Tensor)
	at ai.djl.inference.Predictor.batchPredict(Predictor.java:192) ~[api-0.26.0.jar:?]
	at ai.djl.serving.wlm.ModelInfo$ModelThread.lambda$run$1(ModelInfo.java:1088) ~[wlm-0.26.0.jar:?]
	at ai.djl.serving.wlm.Job.runAll(Job.java:68) ~[wlm-0.26.0.jar:?]
	at ai.djl.serving.wlm.ModelInfo$ModelThread.run(ModelInfo.java:1088) ~[wlm-0.26.0.jar:?]
	at ai.djl.serving.wlm.WorkerThread.runJobs(WorkerThread.java:137) ~[wlm-0.26.0.jar:?]
	at ai.djl.serving.wlm.WorkerThread.run(WorkerThread.java:87) [wlm-0.26.0.jar:?]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.base/java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: ai.djl.engine.EngineException: Expected at most 2 argument(s) for operator 'forward', but received 3 argument(s). Declaration: forward(__torch__.sentence_transformers.SentenceTransformer.SentenceTransformer self, Dict(str, Tensor) input) -> Dict(str, Tensor)
	at ai.djl.pytorch.jni.PyTorchLibrary.moduleRunMethod(Native Method) ~[pytorch-engine-0.26.0.jar:?]
	at ai.djl.pytorch.jni.IValueUtils.forward(IValueUtils.java:57) ~[pytorch-engine-0.26.0.jar:?]
	at ai.djl.pytorch.engine.PtSymbolBlock.forwardInternal(PtSymbolBlock.java:146) ~[pytorch-engine-0.26.0.jar:?]
	at ai.djl.nn.AbstractBaseBlock.forward(AbstractBaseBlock.java:79) ~[api-0.26.0.jar:?]
	at ai.djl.nn.Block.forward(Block.java:127) ~[api-0.26.0.jar:?]
	at ai.djl.inference.Predictor.predictInternal(Predictor.java:143) ~[api-0.26.0.jar:?]
	at ai.djl.inference.Predictor.batchPredict(Predictor.java:170) ~[api-0.26.0.jar:?]
	... 10 more) ~[api-0.26.0.jar:?]
	at ai.djl.inference.Predictor.batchPredict(Predictor.java:170) ~[api-0.26.0.jar:?]
	... 10 more
```
I've probably done something dumb here.. any ideas what?  The input predictor data I believe is as expected, and is handled basically this sort of code:

```java
    private final HuggingFaceTokenizer tokenizer;
        HuggingFaceTokenizer.newInstance("sentence-transformers/clip-ViT-B-32-multilingual-v1");

    @Override
    public NDList processInput(TranslatorContext ctx, Input input)
    {
        Encoding encoding = tokenizer.encode(input.getAsString(0));
        return encoding.toNDList(ctx.getNDManager(), false);
    }
```

--- a/extensions/tokenizers/src/main/java/ai/djl/huggingface/translator/TextEmbeddingTranslator.java
+++ b/extensions/tokenizers/src/main/java/ai/djl/huggingface/translator/TextEmbeddingTranslator.java
@@ -12,18 +12,24 @@
  */
 package ai.djl.huggingface.translator;
 
+import ai.djl.Device;
 import ai.djl.huggingface.tokenizers.Encoding;
 import ai.djl.huggingface.tokenizers.HuggingFaceTokenizer;
 import ai.djl.ndarray.NDArray;
 import ai.djl.ndarray.NDList;
 import ai.djl.ndarray.NDManager;
 import ai.djl.ndarray.types.DataType;
+import ai.djl.ndarray.types.Shape;
 import ai.djl.translate.ArgumentsUtil;
 import ai.djl.translate.Batchifier;
 import ai.djl.translate.Translator;
 import ai.djl.translate.TranslatorContext;
 
 import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.Map;
 
 /** The translator for Huggingface text embedding model. */
@@ -36,24 +42,63 @@ public class TextEmbeddingTranslator implements Translator<String, float[]> {
     private boolean normalize;
     private String pooling;
     private boolean includeTokenTypes;
+    private String dense;
+    private String denseActivation;
+    private String layerNorm;
+    private NDList denseModel;
+    private NDList layerNormModel;
 
     TextEmbeddingTranslator(
             HuggingFaceTokenizer tokenizer,
             Batchifier batchifier,
             String pooling,
             boolean normalize,
-            boolean includeTokenTypes) {
+            boolean includeTokenTypes,
+            String dense,
+            String denseActivation,
+            String layerNorm) {
         this.tokenizer = tokenizer;
         this.batchifier = batchifier;
         this.pooling = pooling;
         this.normalize = normalize;
         this.includeTokenTypes = includeTokenTypes;
+        this.dense = dense;
+        this.denseActivation = denseActivation;
+        this.layerNorm = layerNorm;
     }
 
     /** {@inheritDoc} */
     @Override
     public Batchifier getBatchifier() {
         return batchifier;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void prepare(TranslatorContext ctx) throws Exception {
+        NDManager manager = ctx.getPredictorManager().newSubManager(Device.cpu());
+        if (dense != null) {
+            Path file = Paths.get(dense);
+            if (!file.isAbsolute()) {
+                file = ctx.getModel().getModelPath().resolve(file);
+            }
+            if (Files.exists(file)) {
+                try (InputStream is = Files.newInputStream(file)) {
+                    denseModel = NDList.decode(manager, is);
+                }
+            }
+        }
+        if (layerNorm != null) {
+            Path file = Paths.get(layerNorm);
+            if (!file.isAbsolute()) {
+                file = ctx.getModel().getModelPath().resolve(file);
+            }
+            if (Files.exists(file)) {
+                try (InputStream is = Files.newInputStream(file)) {
+                    layerNormModel = NDList.decode(manager, is);
+                }
+            }
+        }
     }
 
     /** {@inheritDoc} */
@@ -70,6 +115,25 @@ public class TextEmbeddingTranslator implements Translator<String, float[]> {
         Encoding encoding = (Encoding) ctx.getAttachment("encoding");
         NDManager manager = ctx.getNDManager();
         NDArray embeddings = processEmbedding(manager, list, encoding, pooling);
+        embeddings = embeddings.toDevice(Device.cpu(), false);
+        if (denseModel != null) {
+            NDArray weight = denseModel.get("linear.weight");
+            NDArray bias = denseModel.get("linear.bias");
+            embeddings = embeddings.getNDArrayInternal().linear(embeddings, weight, bias).get(0);
+            if ("Tanh".equals(denseActivation)) {
+                embeddings = embeddings.tanh();
+            }
+        }
+        if (layerNormModel != null) {
+            NDArray weight = layerNormModel.get("norm.weight");
+            NDArray bias = layerNormModel.get("norm.bias");
+            Shape shape = weight.getShape();
+            embeddings =
+                    embeddings
+                            .getNDArrayInternal()
+                            .layerNorm(embeddings, shape, weight, bias, 1e-5f)
+                            .get(0);
+        }
         if (normalize) {
             embeddings = embeddings.normalize(2, 0);
         }
@@ -176,6 +240,9 @@ public class TextEmbeddingTranslator implements Translator<String, float[]> {
         private boolean normalize = true;
         private String pooling = "mean";
         private boolean includeTokenTypes;
+        private String dense;
+        private String denseActivation;
+        private String layerNorm;
 
         Builder(HuggingFaceTokenizer tokenizer) {
             this.tokenizer = tokenizer;
@@ -235,6 +302,39 @@ public class TextEmbeddingTranslator implements Translator<String, float[]> {
         }
 
         /**
+         * Sets the dense layer model file for the {@link Translator}.
+         *
+         * @param dense path to dense layer model file
+         * @return this builder
+         */
+        public Builder optDense(String dense) {
+            this.dense = dense;
+            return this;
+        }
+
+        /**
+         * Sets the dense activation function for the {@link Translator}.
+         *
+         * @param denseActivation path to dense layer
+         * @return this builder
+         */
+        public Builder optDenseActivation(String denseActivation) {
+            this.denseActivation = denseActivation;
+            return this;
+        }
+
+        /**
+         * Sets the LayerNorm model for the {@link Translator}.
+         *
+         * @param layerNorm path to LayerNorm model
+         * @return this builder
+         */
+        public Builder optLayerNorm(String layerNorm) {
+            this.layerNorm = layerNorm;
+            return this;
+        }
+
+        /**
          * Configures the builder with the model arguments.
          *
          * @param arguments the model arguments
@@ -245,6 +345,9 @@ public class TextEmbeddingTranslator implements Translator<String, float[]> {
             optNormalize(ArgumentsUtil.booleanValue(arguments, "normalize", true));
             optPoolingMode(ArgumentsUtil.stringValue(arguments, "pooling", "mean"));
             optIncludeTokenTypes(ArgumentsUtil.booleanValue(arguments, "includeTokenTypes"));
+            optDense(ArgumentsUtil.stringValue(arguments, "dense"));
+            optDenseActivation(ArgumentsUtil.stringValue(arguments, "denseActivation"));
+            optLayerNorm(ArgumentsUtil.stringValue(arguments, "layerNorm"));
         }
 
         /**
@@ -255,7 +358,14 @@ public class TextEmbeddingTranslator implements Translator<String, float[]> {
          */
         public TextEmbeddingTranslator build() throws IOException {
             return new TextEmbeddingTranslator(
-                    tokenizer, batchifier, pooling, normalize, includeTokenTypes);
+                    tokenizer,
+                    batchifier,
+                    pooling,
+                    normalize,
+                    includeTokenTypes,
+                    dense,
+                    denseActivation,
+                    layerNorm);
         }
     }
 }

--- a/extensions/tokenizers/src/main/python/fill_mask_converter.py
+++ b/extensions/tokenizers/src/main/python/fill_mask_converter.py
@@ -59,5 +59,6 @@ class FillMaskConverter(HuggingfaceConverter):
         text = self.inputs.replace("[MASK]", tokenizer.mask_token)
         return tokenizer.encode_plus(text, return_tensors='pt')
 
-    def get_extra_arguments(self, hf_pipeline, model_id: str) -> dict:
+    def get_extra_arguments(self, hf_pipeline, model_id: str,
+                            temp_dir: str) -> dict:
         return {"maskToken": hf_pipeline.tokenizer.mask_token}

--- a/extensions/tokenizers/src/main/python/huggingface_converter.py
+++ b/extensions/tokenizers/src/main/python/huggingface_converter.py
@@ -10,7 +10,6 @@
 # or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS"
 # BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied. See the License for
 # the specific language governing permissions and limitations under the License.
-
 import logging
 import os.path
 import shutil
@@ -142,14 +141,15 @@ class HuggingfaceConverter:
 
         # Save serving.properties
         serving_file = os.path.join(temp_dir, "serving.properties")
-        arguments = self.get_extra_arguments(hf_pipeline, model_id)
+        arguments = self.get_extra_arguments(hf_pipeline, model_id, temp_dir)
+        if include_types:
+            arguments["includeTokenTypes"] = "true"
+        arguments["translatorFactory"] = self.translator
+
         with open(serving_file, 'w') as f:
             f.write(f"engine=PyTorch\n"
                     f"option.modelName={model_name}\n"
-                    f"option.mapLocation=true\n"
-                    f"translatorFactory={self.translator}\n")
-            if include_types:
-                f.write(f"includeTokenTypes={include_types}\n")
+                    f"option.mapLocation=true\n")
 
             for k, v in arguments.items():
                 f.write(f"{k}={v}\n")
@@ -160,10 +160,11 @@ class HuggingfaceConverter:
         zip_dir(temp_dir, zip_file)
 
         # Save metadata.json
+        arguments["engine"] = "PyTorch"
         sha1 = sha1_sum(zip_file)
         file_size = os.path.getsize(zip_file)
-        metadata = HuggingfaceMetadata(model_info, self.application,
-                                       self.translator, sha1, file_size)
+        metadata = HuggingfaceMetadata(model_info, self.application, sha1,
+                                       file_size, arguments)
         metadata_file = os.path.join(repo_dir, "metadata.json")
         metadata.save_metadata(metadata_file)
 
@@ -205,7 +206,8 @@ class HuggingfaceConverter:
 
         return self.verify_jit_output(hf_pipeline, encoding, out)
 
-    def get_extra_arguments(self, hf_pipeline, model_id: str) -> dict:
+    def get_extra_arguments(self, hf_pipeline, model_id: str,
+                            temp_dir: str) -> dict:
         return {}
 
     def verify_jit_output(self, hf_pipeline, encoding, out):

--- a/extensions/tokenizers/src/main/python/metadata.py
+++ b/extensions/tokenizers/src/main/python/metadata.py
@@ -16,15 +16,15 @@ from huggingface_models import get_lang_tags
 
 class HuggingfaceMetadata:
 
-    def __init__(self, model_info, application: str, translator: str,
-                 sha1: str, file_size: int):
+    def __init__(self, model_info, application: str, sha1: str, file_size: int,
+                 arguments: dict):
         self.model_info = model_info
         self.artifact_id = model_info.modelId
         self.model_name = model_info.modelId.split("/")[-1]
         self.application = application
-        self.translator = translator
         self.sha1 = sha1
         self.file_size = file_size
+        self.arguments = arguments
 
     def save_metadata(self, metadata_file: str):
         properties = get_lang_tags(self.model_info)
@@ -57,10 +57,7 @@ class HuggingfaceMetadata:
                 "snapshot": False,
                 "name": self.model_name,
                 "properties": properties,
-                "arguments": {
-                    "engine": "PyTorch",
-                    "translatorFactory": self.translator
-                },
+                "arguments": self.arguments,
                 "options": {
                     "mapLocation": True
                 },

--- a/extensions/tokenizers/src/main/python/sentence_similarity_converter.py
+++ b/extensions/tokenizers/src/main/python/sentence_similarity_converter.py
@@ -13,6 +13,7 @@
 import json
 import logging
 import os
+import shutil
 
 import requests
 import torch
@@ -57,25 +58,88 @@ class SentenceSimilarityConverter(HuggingfaceConverter):
 
         return True, None
 
-    def get_extra_arguments(self, hf_pipeline, model_id: str) -> dict:
+    def get_extra_arguments(self, hf_pipeline, model_id: str,
+                            temp_dir: str) -> dict:
         args = {"padding": "true"}
+        pooling_path = None
+        dense_path = None
+        layer_norm_path = None
+        normalize = False
         try:
-            file = hf_hub_download(repo_id=model_id,
-                                   filename="1_Pooling/config.json")
-            if os.path.exists(file):
-                with open(file, "r") as f:
-                    pooling = json.load(f)
-                    if pooling.get("pooling_mode_cls_token"):
-                        args["pooling"] = "cls"
-                    elif pooling.get("pooling_mode_max_tokens"):
-                        args["pooling"] = "max"
-                    elif pooling.get("pooling_mode_mean_sqrt_len_tokens"):
-                        args["pooling"] = "mean_sqrt_len"
-                    elif pooling.get("pooling_mode_weightedmean_tokens"):
-                        args["pooling"] = "weightedmean"
-                    elif pooling.get("pooling_mode_lasttoken"):
-                        args["pooling"] = "lasttoken"
+            file = hf_hub_download(repo_id=model_id, filename="modules.json")
+            with open(file, "r") as f:
+                modules = json.load(f)
+
+            for module in modules:
+                module_type = module.get("type")
+                if module_type == "sentence_transformers.models.Pooling":
+                    pooling_path = module["path"]
+                elif module_type == "sentence_transformers.models.Dense":
+                    dense_path = module["path"]
+                elif module_type == "sentence_transformers.models.LayerNorm":
+                    layer_norm_path = module["path"]
+                elif module_type == "sentence_transformers.models.Normalize":
+                    normalize = "true"
+                elif module_type != "sentence_transformers.models.Transformer":
+                    logging.warning(f"Unexpected module: {module_type}.")
         except requests.exceptions.HTTPError:
-            logging.warning(f"{model_id}: 1_Pooling/config.json not found.")
+            logging.warning(f"{model_id}: modules.json not found.")
+
+        if pooling_path:
+            try:
+                file = hf_hub_download(repo_id=model_id,
+                                       filename=f"{pooling_path}/config.json")
+                if os.path.exists(file):
+                    with open(file, "r") as f:
+                        pooling = json.load(f)
+                        if pooling.get("pooling_mode_cls_token"):
+                            args["pooling"] = "cls"
+                        elif pooling.get("pooling_mode_max_tokens"):
+                            args["pooling"] = "max"
+                        elif pooling.get("pooling_mode_mean_sqrt_len_tokens"):
+                            args["pooling"] = "mean_sqrt_len"
+                        elif pooling.get("pooling_mode_weightedmean_tokens"):
+                            args["pooling"] = "weightedmean"
+                        elif pooling.get("pooling_mode_lasttoken"):
+                            args["pooling"] = "lasttoken"
+            except requests.exceptions.HTTPError:
+                logging.warning(
+                    f"{model_id}: {pooling_path}/config.json not found.")
+
+        if dense_path:
+            try:
+                file = hf_hub_download(repo_id=model_id,
+                                       filename=f"{dense_path}/config.json")
+                with open(file, "r") as f:
+                    dense = json.load(f)
+                    activation = dense.get("activation_function")
+                    if activation == "torch.nn.modules.activation.Tanh":
+                        args["denseActivation"] = "Tanh"
+                    elif activation != "torch.nn.modules.linear.Identity":
+                        logging.warning(
+                            f"Unexpected activation function: {activation}.")
+                self.save_module_weight(model_id, temp_dir, dense_path,
+                                        "linear")
+                args["dense"] = "linear.safetensors"
+            except requests.exceptions.HTTPError:
+                logging.debug(f"{model_id}: {dense_path} not found.")
+
+        if layer_norm_path:
+            try:
+                self.save_module_weight(model_id, temp_dir, layer_norm_path,
+                                        "norm")
+                args["layerNorm"] = "norm.safetensors"
+            except requests.exceptions.HTTPError:
+                logging.warning(f"{model_id}: {layer_norm_path} not found.")
+
+        if not normalize:
+            args["normalize"] = "false"
 
         return args
+
+    @staticmethod
+    def save_module_weight(model_id: str, temp_dir: str, layer: str,
+                           name: str):
+        file = hf_hub_download(repo_id=model_id,
+                               filename=f"{layer}/model.safetensors")
+        shutil.copyfile(file, os.path.join(temp_dir, f"{name}.safetensors"))

